stairway/src/main/java/bio/terra/stairway/queue/QueueMessageReady.java

```diff
@@ -16,18 +19,24 @@ class QueueMessageReady extends QueueMessage {
 
   private QueueMessageType type;
   private String flightId;
+  private Map<String, String> callingThreadContext;
 
   private QueueMessageReady() {}
 
   public QueueMessageReady(String flightId) {
     this.type =
         new QueueMessageType(QueueMessage.FORMAT_VERSION, QueueMessageEnum.QUEUE_MESSAGE_READY);
     this.flightId = flightId;
+    this.callingThreadContext = MDC.getCopyOfContextMap();
   }
 
   @Override
   public boolean process(StairwayImpl stairwayImpl) throws InterruptedException {
+    boolean processed = false;
+    // Save the initial thread context so that it can be restored
+    Map<String, String> initialContext = MDC.getCopyOfContextMap();
     try {
+      MdcUtils.overwriteContext(callingThreadContext);
```

Comment:
Could this be done inside `McdUtils`? Something like
```
    MdcUtils.callWithContext(callingThreadContext, () -> {
        ...
        stairwayImpl.resume(flightId)
        ...
    });
```

I think it would be better to not expose less of the inner workings of `MdcUtils` if possible. And there may be other cases where we want to overwrite the context temporarily like this.

Reply:
Thanks for the suggestion. I added new public utility method `MdcUtils.callWithContext` and removed the `public` modifier on `MdcUtils.overwriteContext`.  Any possible refactoring of other code to use this method I'll call out of scope.